src/config/source.rs

```diff
@@ -15,6 +16,19 @@ const ALL: [Source; 3] = [Source::Json, Source::Yml, Source::GraphQL];
 #[error("Unsupported file extension: {0}")]
 pub struct UnsupportedFileFormat(String);
 
+impl std::str::FromStr for Source {
+  type Err = String;
+
+  fn from_str(s: &str) -> Result<Self, Self::Err> {
+    match s.to_lowercase().as_str() {
+      "json" => Ok(Source::Json),
+      "yml" | "yaml" => Ok(Source::Yml),
+      "graphql" | "gql" => Ok(Source::GraphQL),
+      _ => Err(format!("Unsupported format: {}", s)),
+    }
+  }
+}
```

Comment:
Do we need this? Can't we use `Source::detect`?

Reply:
Looks like "detect" is using Souce::ends with that checks for an `.yml`, `.graphql` and `.json` (with a dot, because it checks for an extension name). It also doesn't take into account things like "gql" or "yaml". Should I modify it or keep it as it is?